File: C/beginner/1045.c
/*https://judge.beecrowd.com/pt/problems/view/1045*/

/*Including libs*/
#include <stdio.h>
#include <math.h>

/*Main Function*/
void swap(double *a, double *b)
{
    double temp = *a;
    *a = *b;
    *b = temp;
}

int main()
{
    double A, B, C;

    scanf("%lf %lf %lf", &A, &B, &C);

    if (B > A)
        swap(&A, &B);
    if (C > B)
    {
        swap(&B, &C);
        if (B > A)
            swap(&A, &B);
    }

    if (A >= B + C)
    {
        printf("NAO FORMA TRIANGULO\n");
    }
    else
    {
        if (A * A == B * B + C * C)
            printf("TRIANGULO RETANGULO\n");
        else if (A * A > B * B + C * C)
            printf("TRIANGULO OBTUSANGULO\n");
        else
            printf("TRIANGULO ACUTANGULO\n");

        if (A == B && B == C)
            printf("TRIANGULO EQUILATERO\n");
        else if (A == B || A == C || B == C)
            printf("TRIANGULO ISOSCELES\n");
    }

    return 0;
}

/*By Fernando Serra*/
/*https://github.com/fernandosserra*/
/*Special thanks to xTecna on https://github.com/xTecna who made the original code*/

File: Haskell/beginner/1045.hs
-- https://judge.beecrowd.com/pt/problems/view/1045 
-- Importing required functions
import Text.Printf
import Data.List (sort)

main :: IO ()
main = do
    -- Requesting input
    input <- getLine
    let lados = map read (words input) :: [Double]
    let [a, b, c] = reverse (sort lados)  -- Sort in descending order

    -- Check if it forms a triangle
    if a >= b + c then
        printf "NAO FORMA TRIANGULO\n"
    else do
        -- Check triangle type
        if a^2 == b^2 + c^2 then
            printf "TRIANGULO RETANGULO\n"
        else if a^2 > b^2 + c^2 then
            printf "TRIANGULO OBTUSANGULO\n"
        else
            printf "TRIANGULO ACUTANGULO\n"

        -- Check if it is equilateral or isosceles
        if a == b && b == c then
            printf "TRIANGULO EQUILATERO\n"
        else if a == b || a == c || b == c then
            printf "TRIANGULO ISOSCELES\n"
        else
            return()

-- By Fernando Serra
-- https://github.com/fernandosserra
-- I had a lot of help from ChatGPT and the Haskell documentation for this fix